### PR TITLE
Add test for sliver_animated_opacity.0.dart API example.

### PR DESCRIPTION
**Pull request by ksokolovskyi**
This PR contributes to https://github.com/flutter/flutter/issues/130459

### Description
- Adds `examples/api/test/widgets/implicit_animations/sliver_animated_opacity.0_test.dart` test

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

File: examples/api/test/widgets/implicit_animations/sliver_animated_opacity.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        find.descendant(
+          of: find.byType(SliverAnimatedOpacity),
+          matching: find.byType(SliverFadeTransition),
+        ),
+      );
+      expect(fadeTransition.opacity.value, 1);
+
+      // Advance animation to the end by the 500ms duration specified in
+      // the example app.
+      await tester.pump(const Duration(milliseconds: 500));
```
Comment (justinmc) [inline examples/api/test/widgets/implicit_animations/sliver_animated_opacity.0_test.dart:39]: Nit: Consider exposing this Duration in the example app and referencing it here.

**Review by justinmc — APPROVED** [review]
LGTM :+1:

**Comment by ksokolovskyi** [comment]
@justinmc @Renzo-Olivares thanks for the review!
I applied suggestions in the latest commit.

**Review by Renzo-Olivares — APPROVED** [review]
LGTM

**Comment by Renzo-Olivares** [comment]
@ksokolovskyi can you try rebasing your PR on `master`? The google testing failures looks unrelated.

**Comment by ksokolovskyi** [comment]
@Renzo-Olivares @justinmc thanks for the review!

**Pull request merged**